tests/js/percentageTests.js

```diff
@@ -0,0 +1,129 @@
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined], // total === ""
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined], // total === "string"
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 100, 100, 0, 0, 2272.7272727272725, 2272.7272727272725, 4545.454545454545, 4545.454545454545], // total === 2.2
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 100, 100, 0, 0, 2272.7272727272725, 2272.7272727272725, 4545.454545454545, 4545.454545454545], // total === "2.2"
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 0, 0, 0, 0, 0, 0, 0, 0], // total === 0
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 0, 0, 0, 0, 0, 0, 0, 0], // total === "0"
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 4.4, 4.4, 0, 0, 100, 100, 200, 200], // total === 50
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 4.4, 4.4, 0, 0, 100, 100, 200, 200], // total === "50"
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 2.2, 2.2, 0, 0, 50, 50, 100, 100], // total === 100
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 2.2, 2.2, 0, 0, 50, 50, 100, 100] // total === "100"
+    ];
+
+    jqUnit.test("Test gpii.chartAuthoring.percentage.calculate", function () {
+        fluid.each(gpii.tests.calculatePercentage.inputs, function (total, totalIdx) {
+            fluid.each(gpii.tests.calculatePercentage.inputs, function (value, valIdx) {
+                var actual = gpii.chartAuthoring.percentage.calculate(value, total);
+                var expected = gpii.tests.calculatePercentage.outputs[totalIdx][valIdx];
+                // Using fluid.model.isSameValue to remove any javascript precision errors on floating point numbers.
+                jqUnit.assertTrue("The percentate for value: " + value + " total: " + total + " should be calculated as " + expected, fluid.model.isSameValue(expected, actual));
+            });
+        });
+    });
+
+    fluid.registerNamespace("gpii.tests.retrieveVal");
+
+    gpii.tests.retrieveVal.percentages = [
+        "10.5",
+        10.5,
+        "10",
+        10,
+        function () {return "10.5";},
```

Comment:
I think that the facility to supply a function returning a value adds unnecessary complexity and not enough value (no pun intended) to the system. In general, the IoC and model relay systems together should provide enough flexibility to deal with situations where the percentage needs to be sourced from non-literal data. I think we should remove the facility and test cases for "function returning value".